quinn/src/connection.rs

```diff
@@ -366,6 +366,26 @@ impl Connection {
         }
     }
 
+    /// Wait for the connection to be closed for any reason
+    pub async fn closed(&self) -> ConnectionError {
```

Comment:
I'm waffling over whether this should return `Result<(), ConnectionError>`, by mapping `ConnectionError::LocallyClosed` to `Ok(())`. It doesn't add any information, but it does make unexpected vs. expected failures a little clearer--except the peer closing a connection is also an "expected" failure and certainly can't be considered `Ok`, so maybe any attempt to interpret for the application's benefit is futile?

Reply:
Yeah, it seems like singling out the `LocallyClosed` doesn't quite make sense. I agree that it looks a little strange to unconditionally return a type with an `Error` suffix, but I guess we should solve that using some additional documentation?

Reply:
Added some brief discussion.